### PR TITLE
Forward to Universal RendererData upgrader fix

### PR DESCRIPTION
**Pull request by cinight**
### Purpose of this PR
The ForwardRendererData to UniversalRendererData upgrader doesn't work when the upgrade changes are being discarded using source control. This PR make sure the upgrader will always auto-upgrade the ForwardRendererData again if that happens.
This PR also fixes https://fogbugz.unity3d.com/f/cases/1332463/

### Testing status
Please see [test doc](https://docs.google.com/document/d/12LxVFFP30pDTk1rrpQqocjeNJvWt5LLqukKIZoDkV4o/edit#). I've manually tested the new and previous test cases.
Update: I've also triggered [URP PR ABV](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix%252FuniversalRendererUpgraderfix/.yamato%252Fall-universal_split.yml%2523PR_Universal_Split_CUSTOM-REVISION/6684357/job/pipeline) against Ono branch `assetpipeline/initial-refresh` f99434ab9296. XR tests failed is due to XR changes in trunk.

### Comments to reviewers
This upgrader fix relies on an OnPostprocessAllAssets bugfix which hasn't land trunk yet. 
So please use the build from the Ono branch `assetpipeline/initial-refresh`.

The only bad thing is that, the "no default renderer" / "deprecated ForwardRendererData" error will still happen when the upgraded changes are discarded in source control, but rendering is not harmed. So the behavior is like:
1. BoatAttack project opens and upgrade done
    - no error + rendering fine
2. Discard changes in source control
3. Back to Unity
     - a few ":error:no default renderer" errors in console but rendering is fine

### CODE REVIEW

**Comment by cinight** [comment]
I still need to work on the script a bit. There are a few improvements and might be an AssetDatabase bug affecting this so I need to try them out.
I'll re-test with all the test cases to make sure the script still covers all the situations.

**Comment by cinight** [comment]
I've retested all the test cases and updated the behavior in PR description.

**Review by phi-lira — APPROVED** [review]

**Comment by cinight** [comment]
I've also triggered URP PR ABV against Ono branch assetpipeline/initial-refresh f99434ab9296. XR tests failed is due to XR changes in trunk. So once the AssetDatabase PR lands trunk I'll merge this PR.

**Review by cdxntchou — APPROVED** [review]

File: com.unity.render-pipelines.universal/Editor/AssetPostProcessors/ForwardToUniversalRendererPostprocessor.cs [hunk 1]
```diff
@@ -31,62 +42,93 @@ static void UpgradeAsset(Object rendererData, string rendererDataPath, Object fw
             }
         }
 
-        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
```
Comment (martint-unity) [inline com.unity.render-pipelines.universal/Editor/AssetPostProcessors/ForwardToUniversalRendererPostprocessor.cs:34]: Is this safe to remove?

**Comment by cinight** [comment]
Going to merge this without the assetpipeline fix, I've re-tested all test cases using current trunk and only URP template will stay as broken as it is now. But URP template is upgraded by https://github.com/Unity-Technologies/Graphics/pull/4486/, and is in Graphics master now.
I also triggered a few URP Yamato tests against current trunk, all green ([foundation on OSX](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix%252FuniversalRendererUpgraderfix/.yamato%252Furp_foundation-osx-metal.yml%2523URP_Foundation_OSX_Metal_playmode_cache_mono_Linear_trunk),[lighting on DX11](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix%252FuniversalRendererUpgraderfix/.yamato%252Furp_lighting-win-dx11.yml%2523URP_Lighting_Win_DX11_editmode_mono_Linear_trunk),[postpro on DX11 XR](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix%252FuniversalRendererUpgraderfix/.yamato%252Furp_postpro-win-dx11.yml%2523URP_PostPro_Win_DX11_playmode_XR_mono_Linear_trunk)). so merging this PR looks safe to me. (discussed on [slack](https://unity.slack.com/archives/G010YL29H8F/p1620822215361800) too)

**Pull request merged**